inst/include/RcppEigenCholmod.h

```diff
@@ -19,1072 +21,64 @@
 // You should have received a copy of the GNU General Public License
 // along with RcppEigen.  If not, see <http://www.gnu.org/licenses/>.
 
-#ifndef RcppEigen_CHOLMOD_H
-#define RcppEigen_CHOLMOD_H
+#ifndef RcppEigen__RcppEigenCholmod__h
+#define RcppEigen__RcppEigenCholmod__h
 
-/* Original of this is Matrix/inst/include/cholmod.h --- sync with it (e.g for lme4) !
-                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
- */
-#include <stddef.h>
-#include <limits.h>
-
-// Rather use C99 -- which we require in R anyway
-#include <inttypes.h>
-
-
-#ifdef	__cplusplus
-extern "C" {
+#include <Matrix.h>
+#ifndef R_MATRIX_CHOLMOD /* Matrix <= 1.6-1.1 */
+# define R_MATRIX_CHOLMOD(_NAME_) M_cholmod_ ## _NAME_
+# define M_cholmod_start M_R_cholmod_start /* sigh */
```

Comment:
(Just a question: Remind again what that second line does / why we need it / why the sigh?  )

Reply:
**Matrix** <= 1.6-1.1 uses the `M_` prefix for all of its stubs except for `cholmod_start`, where it uses the `M_R_` prefix.  1.6-2 uses `M_` everywhere and preserves backwards compatibility with the same `#define`.  A minor nuisance ...

Reply:
Ahhhh. A local fix.  I see now!